marconi-chain-index/src/Marconi/ChainIndex/Indexers/Utxo.hs

```diff
@@ -286,37 +333,57 @@ instance Buffered UtxoHandle where
          (unless
           (null rows)
           (SQL.executeMany c
-            [r|INSERT OR REPLACE INTO unspent_transactions
-                ( address, txId, txIx, datum, datumHash
-                , value, inlineScript, inlineScriptHash
-                , slotNo, blockHash
-                ) VALUES (?,?,?,?,?,?,?,?,?,?)|] rows))
+            [r|INSERT
+               OR REPLACE INTO unspent_transactions (
+                 address,
+                 txId,
+                 txIx,
+                 datum,
+                 datumHash,
+                 value,
+                 inlineScript,
+                 inlineScriptHash,
+                 slotNo,
+                 blockHash
+              ) VALUES
+              (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)|] rows))
          (unless
           (null spents)
           (SQL.executeMany c
-           [r|INSERT OR REPLACE INTO spent
-               ( txInTxId, txInTxIx, slotNo, blockHash
-               ) VALUES (?,?,?,?)|] spents)))
+           [r|INSERT
+              OR REPLACE INTO spent (
+                txId,
+                txIx, slotNo, blockHash
+              ) VALUES
+              (?, ?, ?, ?)|] spents)))
     -- We want to perform vacuum about once every 100 * buffer ((k + 1) * 2)
-    rndCheck <- createSystemRandom >>= uniformR (1 :: Int, 100)
-    when (rndCheck == 42) $ do
-      SQL.execute_ c [r|DELETE FROM unspent_transactions
-                          WHERE unspent_transactions.rowid IN
-                            (SELECT unspent_transactions.rowid
-                             FROM unspent_transactions
-                               LEFT JOIN spent ON
-                                 unspent_transactions.txId = spent.txInTxId
-                               AND unspent_transactions.txIx = spent.txInTxIx
-                             WHERE spent.txInTxId IS NOT NULL)|]
-      SQL.execute_ c "VACUUM"
+    when (toVacuume h) $ do
```

Comment:
The comment is incorrect: we do it randomly on a mean of once every 100.
As discussed, we need to be more cautious about it, especially when disk contains events that can be rollbacked but may be not in this PR (we can at least use `toVacuume = False` now)

Reply:
good catch, thank you. I fixed the comment.
 I agree that if we're going to vacuum, we need more intelligent way of doing it.
I also have no objection to setting the flag `toVaccume=False` by default, unless the user instruct us otherwise through CLI params that yet to be added!